tests/unit/plugins/modules/service/test_service.py

```diff
@@ -0,0 +1,333 @@
+
+    with pytest.raises(AnsibleFailJson, match="Cluster does not exist"):
+        service.main()
+
+
+def test_present_missing_type(conn, module_args):
+    conn.update(
+        cluster=os.getenv("CM_CLUSTER"),
+        service=os.getenv("CM_SERVICE"),
+    )
+    module_args(conn)
+
+    with pytest.raises(AnsibleFailJson, match="type"):
+        service.main()
+
+
+def test_present_create_service(conn, module_args):
+    conn.update(
+        cluster=os.getenv("CM_CLUSTER"),
+        service=os.getenv("CM_SERVICE"),
+        display_name="Example Service",
+        type="ZOOKEEPER",
+    )
+    module_args(conn)
+
+    with pytest.raises(AnsibleExitJson) as e:
+        service.main()
+
+    assert e.value.changed == True
+
+    with pytest.raises(AnsibleExitJson) as e:
+        service.main()
+
+    assert e.value.changed == False
+
+
+def test_present_update_service(conn, module_args):
+    conn.update(
+        cluster=os.getenv("CM_CLUSTER"),
+        service=os.getenv("CM_SERVICE"),
+        display_name="Example Service by Ansible",
+    )
+    module_args(conn)
+
+    with pytest.raises(AnsibleExitJson) as e:
+        service.main()
+
+    assert e.value.changed == True
+
+    with pytest.raises(AnsibleExitJson) as e:
+        service.main()
+
+    assert e.value.changed == False
+
+
+def test_present_maintenance_mode(conn, module_args):
+    conn.update(
+        cluster=os.getenv("CM_CLUSTER"),
+        service=os.getenv("CM_SERVICE"),
+        maintenance="yes",
+    )
+    module_args(conn)
+
+    with pytest.raises(AnsibleExitJson) as e:
+        service.main()
+
+    assert e.value.service["maintenance_mode"] == True
+
+    with pytest.raises(AnsibleExitJson) as e:
+        service.main()
+
+    assert e.value.service["maintenance_mode"] == True
+    assert e.value.changed == False
+
+    conn.update(
+        maintenance="no",
+    )
+    module_args(conn)
+
+    with pytest.raises(AnsibleExitJson) as e:
+        service.main()
+
+    assert e.value.service["maintenance_mode"] == False
+
+    with pytest.raises(AnsibleExitJson) as e:
+        service.main()
+
+    assert e.value.service["maintenance_mode"] == False
+    assert e.value.changed == False
+
+
+def test_present_set_tags(conn, module_args):
+    conn.update(
+        cluster=os.getenv("CM_CLUSTER"),
+        service=os.getenv("CM_SERVICE"),
+        tags=dict(
+            test="Ansible", key="Value", empty_string="", blank_string="  ", none=None
+        ),
+    )
+    module_args(conn)
+
+    with pytest.raises(AnsibleExitJson) as e:
+        service.main()
+
+    assert (
+        recursive_diff(e.value.service["tags"], dict(test="Ansible", key="Value"))
+        is None
+    )
+
+    with pytest.raises(AnsibleExitJson) as e:
+        service.main()
+
+    assert (
+        recursive_diff(e.value.service["tags"], dict(test="Ansible", key="Value"))
+        is None
+    )
+    assert e.value.changed == False
+
+
+def test_present_append_tags(conn, module_args):
```

Comment:
Same here when there is a custom tag added before ,it will append the tags , but it will give back `AssertionError`